src/aztec3/circuits/kernel/private/c_bind.cpp

```diff
@@ -0,0 +1,37 @@
+#include "c_bind.h"
+#include "private_kernel_circuit.hpp"
+#include <aztec3/circuits/abis/tx_request.hpp>
+
+#include <common/streams.hpp>
+#include <common/mem.hpp>
+#include <common/container.hpp>
+#include <cstdint>
+#include <ecc/curves/grumpkin/grumpkin.hpp>
+#include <crypto/pedersen_commitment/pedersen.hpp>
+#include <sstream>
+
+using namespace barretenberg;
+using namespace aztec3::circuits::abis;
+using namespace aztec3::circuits::kernel::private_kernel;
```

Comment:
I really want to avoid any `using namespace <entire namespace>` syntax in this repo. It clutters things greatly, and makes nagivation quite difficult.

Please would we hone in on what we actually need from those namespaces, and specify those things exactly. It's more verbose, but it's much easier to understand what's being used.

Reply:
Got it :+1:  Will do

Reply:
I believe I have addressed this in all files. Although I'm still not sure when it makes sense to put these `using *` lines inside an unnamed `namespace {...}`.

https://github.com/AztecProtocol/aztec3-circuits/pull/26/files#diff-5e95da82d436268a30fac8cc208b926ed4412e86fca4a99025b3439f39eaae4aR7

Reply:
A quick attempt at explaining (I think this is right... mostly...)

```c++
// file a.hpp:

namespace a {
class A {

};
} // namespace a

//-----------------------------------------

// file b.hpp:
#include "a.hpp"

namespace b {

using namespace a; // please don't do `using namespace` anywhere. We should try to add angry linting about it. It will import all names from namespace an into all of namespace b (all files), potentially causing collisions with the names in namespace b, or causing the wrong name to be looked-up. It also makes it harder to know where a name comes from, when you're new to the codebase.

}


//-----------------------------------------

// file c.hpp:
#include "a.hpp"

namespace c {

using a::A; // imports a::A into namespace b (all files). Now `A` refers to a::A in namespace c (all files).

// If there's already a declaration of a `A` in namespace c, then `a::A` will still need to be qualified with a path `a::A.

}

//-----------------------------------------

// file d.hpp:
#include "a.hpp"

// unnamed namespace:
namespace {
using a::A; // `A` is only understood in this file. It does not become understood by namespace d in other files.
}

namespace d {

// If there's already a declaration of a `A` in namespace d, then `a::A` will still need to be qualified with a path `a::A.

}


```

Reply:
Thanks @iAmMichaelConnor this is very helpful